android/app/src/main/java/org/jitsi/meet/AudioModeModule.java

```diff
@@ -0,0 +1,440 @@
+    }
+
+    /**
+     * Helper class to detect and handle Bluetooth device changes.  It monitors Bluetooth headsets
+     * being connected / disconnected and notifies the module about device changes when this occurs.
+     */
+    private class BluetoothManager {
+        /**
+         * Reference to the Bluetooth adapter, needed for managing <tt>BluetoothProfile.HEADSET</tt>
+         * devices.
+         */
+        private BluetoothAdapter bluetoothAdapter;
+
+        /**
+         * Reference to a proxy object which allows us to query connected devices.
+         */
+        private BluetoothHeadset bluetoothHeadset;
+
+        /**
+         * Listener for Bluetooth service profiles, allows us to get the proxy object to
+         * {@link BluetoothHeadset}.
+         */
+        private BluetoothProfile.ServiceListener bluetoothProfileListener;
+
+        /**
+         * Helper for running Bluetooth operations on the main thread.
+         */
+        private Runnable bluetoothRunnable;
+
+        /**
+         * Flag indicating if there are any Bluetooth headset devices currently available.
+         */
+        private boolean headsetAvailable = false;
+
+        /**
+         * Start monitoring Bluetooth device activity.
+         */
+        public void start() {
+            ReactContext reactContext = getReactApplicationContext();
+            if (reactContext == null) {
+                return;
+            }
+
+            bluetoothRunnable = new Runnable() {
+                @Override
+                public void run() {
+                    if (bluetoothHeadset == null) {
+                        headsetAvailable = false;
+                    } else {
+                        List<BluetoothDevice> devices = bluetoothHeadset.getConnectedDevices();
+                        headsetAvailable = !devices.isEmpty();
+                    }
+                    onAudioDeviceChange();
+                }
+            };
+
+            bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+            if (bluetoothAdapter == null) {
+                Log.w(TAG, "Device doesn't support Bluetooth");
+                return;
+            }
+
+            if (!audioManager.isBluetoothScoAvailableOffCall()) {
+                Log.w(TAG, "Bluetooth SCO is not available");
+                return;
+            }
+
+            bluetoothProfileListener = new BluetoothProfile.ServiceListener() {
+                @Override
+                public void onServiceConnected(int profile, BluetoothProfile proxy) {
+                    if (profile == BluetoothProfile.HEADSET) {
+                        bluetoothHeadset = (BluetoothHeadset) proxy;
+                        updateDevices();
+                    }
+                }
+
+                @Override
+                public void onServiceDisconnected(int profile) {
+                    if (profile == BluetoothProfile.HEADSET) {
+                        bluetoothHeadset = null;
```

Comment:
`onServiceDisconnected` contains pretty much the same source code as `onServiceConnected ` with the only difference being the value assigned to `bluetoothHeadset`. Please extract the 2 very similar implementations in 1 function in order to reduce the duplication.